### PR TITLE
fix(chore): remove zgc in dockerfile for ARM env

### PR DESCRIPTION
**Pull request by aroundabout**
## Purpose of the PR

On macos(ARM arch M1/2/3..), zgc is only available with java > 13, so we remove -g zgc in the docker-entrypoint.sh.

img_v3_0278_dd825afb-a548-4fa8-81e6-dd4fe56274bg
img_v3_0278_7d9062b4-7f0f-4911-a50c-896f29d7f38g
P4NlOlI8VR

## Todo

- [x] move init-store to docker-entrypoint
- [x] simply auth and password to auth
- [x] read version from pom

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [ ]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh [hunk 1]
```diff
@@ -34,6 +34,7 @@ else
 fi
 
 # start hugegraph
-./bin/start-hugegraph.sh -j "$JAVA_OPTS" -g zgc
+# remove -g zgc, which is only available on mac with java > 13
```
Comment (imbajin) [inline hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh:37]:
```suggestion
# remove "-g zgc" now, which is only available on ARM-Mac with java > 13 
```

File: hugegraph-server/Dockerfile [hunk 2]
```diff
@@ -26,6 +26,7 @@ ARG MAVEN_ARGS
 RUN mvn package $MAVEN_ARGS -e -B -ntp -DskipTests -Dmaven.javadoc.skip=true && pwd && ls -l
 
 # 2nd stage: runtime env
+# remove zgc, which is only available on mac with java > 13
```
Comment (imbajin) [inline hugegraph-server/Dockerfile:29]:
```suggestion
# remove zgc, which is only available on ARM-Mac with java > 13 
```

File: hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh
Comment (imbajin) [inline hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh:1]: Also add a comment on `*hugegraph.sh` in dist?  "Comment for ZGC"

Comment (aroundabout) [reply hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh:1]: added in hugegraph-server.sh.

**Review by imbajin — DISMISSED** [review]

File: hugegraph-server/hugegraph-dist/src/assembly/static/bin/enable-auth.sh [hunk 3]
```diff
@@ -36,19 +36,21 @@ GRAPH_CONF="hugegraph.properties"
 
 # make a backup
 BAK_CONF="$TOP/conf-bak"
-mkdir -p "$BAK_CONF"
-cp "${CONF}/${GREMLIN_SERVER_CONF}" "${BAK_CONF}/${GREMLIN_SERVER_CONF}.bak"
-cp "${CONF}/${REST_SERVER_CONF}" "${BAK_CONF}/${REST_SERVER_CONF}.bak"
-cp "${CONF}/graphs/${GRAPH_CONF}" "${BAK_CONF}/${GRAPH_CONF}.bak"
+if [ ! -d "$BAK_CONF" ]; then
```
Comment (aroundabout) [inline hugegraph-server/hugegraph-dist/src/assembly/static/bin/enable-auth.sh:39]: new test image is pushed as dandelionivy/server

File: hugegraph-server/Dockerfile [hunk 4]
```diff
@@ -24,12 +24,13 @@ WORKDIR /pkg
 ARG MAVEN_ARGS
 
 RUN mvn package $MAVEN_ARGS -e -B -ntp -DskipTests -Dmaven.javadoc.skip=true && pwd && ls -l
+RUN find . -name 'apache-hugegraph-incubating-*.tar.gz' -delete
```
Comment (imbajin) [inline hugegraph-server/Dockerfile:27]: what's for?

File: hugegraph-server/Dockerfile [hunk 5]
```diff
@@ -24,12 +24,13 @@ WORKDIR /pkg
...
 
 # 2nd stage: runtime env
+# remove zgc, which is only available on ARM-Mac with java > 13
```
Comment (imbajin) [inline hugegraph-server/Dockerfile:30]: seems the comment line could be better ↓

**Review by imbajin — COMMENTED** [review]
THX, and we need  pay attention to the size of the image, and then you can build a binary package

File: hugegraph-server/Dockerfile [hunk 6]
```diff
@@ -48,8 +48,10 @@ RUN set -x \
        curl \
        lsof \
        vim \
+       cron \
     && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+    && rm -rf /var/lib/apt/lists/* \
+    && service cron start
 
 # 2. Init HugeGraph Sever
```
Comment (imbajin) [inline hugegraph-server/Dockerfile:56]: image

maybe step2 could also be merged into st1?

**Review by imbajin — APPROVED** [review]
I'll try it soon

File: hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh [hunk 7]
```diff
@@ -16,24 +16,30 @@
...
-    # set password if use do not provide
+INIT_FLAG_FILE="init_complete"
+
+if [ ! -f "${DOCKER_FOLDER}/${INIT_FLAG_FILE}" ]; then
+    # wait for storage backend
+    ./bin/wait-storage.sh
     if [ -z "$PASSWORD" ]; then
-        echo "you have not set the password, we will use the default password"
-        PASSWORD="hugegraph"
+        ./bin/init-store.sh
```
Comment (imbajin) [inline hugegraph-server/hugegraph-dist/docker/docker-entrypoint.sh:29]: add echo/print before normal-init like "init in non-auth mode"

**Review by imbajin — APPROVED** [review]

**Review by simon824 — APPROVED** [review]
+1

**Pull request merged**